src/modules/job-manager/jobtap.c

```diff
@@ -219,6 +183,216 @@ static flux_plugin_arg_t *jobtap_args_vcreate (struct jobtap *jobtap,
+                          jobtap_error_t *errp)
+{
+    int count = 0;
+    bool isglob = isa_glob (arg);
+    bool all = strcmp (arg, "all") == 0;
+
+    flux_plugin_t *p = zlistx_first (jobtap->plugins);
+    while (p) {
+        const char *name = jobtap_plugin_name (p);
+        if (all
+            || (isglob && fnmatch (arg, name, 0) == 0)
+            || strcmp (arg, name) == 0) {
+            zlistx_detach_cur (jobtap->plugins);
+            flux_plugin_destroy (p);
+            count++;
+        }
+        p = zlistx_next (jobtap->plugins);
+    }
+    if (count == 0 && !all) {
+        errno = ENOENT;
+        return errprintf (errp, "Failed to find plugin to remove");
+    }
+    return count;
+}
+
+static int jobtap_conf_entry (struct jobtap *jobtap,
+                              int index,
+                              json_t *entry,
+                              jobtap_error_t *errp)
+{
+    json_error_t json_err;
+    jobtap_error_t jobtap_err;
+    const char *load = NULL;
+    const char *remove = NULL;
+    json_t *conf = NULL;
+
+    if (json_unpack_ex (entry, &json_err, 0,
+                        "{s?:s s?:o s?:s}",
+                        "load", &load,
+                        "conf", &conf,
+                        "remove", &remove) < 0) {
+        return errprintf (errp,
+                          "[job-manager.plugins][%d]: %s",
+                          index,
+                          json_err.text);
+    }
+    if (remove && jobtap_remove (jobtap, remove, &jobtap_err) < 0) {
+        return errprintf (errp,
+                          "[job-manager.plugins][%d]: remove %s: %s",
+                          index,
+                          remove,
+                          jobtap_err.text);
+    }
+    if (load && jobtap_load_plugin (jobtap,
+                                    load,
+                                    conf,
+                                    &jobtap_err) < 0) {
+        return errprintf (errp,
+                          "[job-manager.plugins][%d]: load %s: %s",
+                          index,
+                          load,
+                          jobtap_err.text);
+    }
+    return 0;
+}
+
+static int jobtap_parse_config (struct jobtap *jobtap,
+                                const flux_conf_t *conf,
+                                jobtap_error_t *errp)
+{
+    json_t *plugins = NULL;
+    flux_conf_error_t error;
+    json_t *entry;
+    int i;
+
+    if (!conf)
+        return errprintf (errp, "conf object can't be NULL");
+
+    if (flux_conf_unpack (conf,
+                          &error,
+                          "{s?:{s?:o}}",
+                          "job-manager",
+                            "plugins", &plugins) < 0) {
+        return errprintf (errp,
+                          "[job-manager.plugins]: %s: %d: %s",
+                          error.filename,
+                          error.lineno,
+                          error.errbuf);
```

Comment:
Not sure filename and lineno are going to be helpful here after file glob -> TOML -> JSON conversion,.

Reply:
Good point.